### PR TITLE
Add string comparison of tones

### PR DESCRIPTION
**Pull request by wangrui-msft**
# Description
Add conversion to string from a collection of tones. 
```
        RecognizeCompleted recognizeCompletedEvent = (RecognizeCompleted) event;
        Optional<RecognizeResult> dtmfResult = recognizeCompletedEvent.getRecognizeResult();
        CollectTonesResult tonesResult = (CollectTonesResult) dtmfResult.get();

        String tones = tonesResult.convertToString();
        if (tones.equals("1234567") {
        // do something
        }


```
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CollectTonesResult.java [hunk 1]
```diff
@@ -36,4 +37,16 @@ public CollectTonesResult setTones(List<DtmfTone> dtmfTones) {
...
     }
+
+    /**
+     * Set the tones property: The tones property.
+     *
+     * @return the CollectTonesResult object itself.
+     */
+    public String convertToString() {
+        return dtmfTones.stream()
+                .map(x -> x.toChar())
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CollectTonesResult.java:48]: why is needed to convert to char? why not to to string directly and you remove one step?

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java [hunk 2]
```diff
@@ -68,8 +68,72 @@ public static DtmfTone fromString(String name) {
         return fromString(name, DtmfTone.class);
     }
 
+    /** @return known Tone values in char. */
+    public char toChar() {
+        return convertToChar();
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:73]: what is the intention of having just a one line method? would it be simpler to add the contents of `convertToChar` in this method?

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java [hunk 3]
```diff
@@ -68,8 +68,72 @@ public static DtmfTone fromString(String name) {
...
+                toneInChar = 'd';
+                break;
+            case "pound":
+                toneInChar = '#';
+                break;
+            case "asterisk":
+                toneInChar = '*';
+                break;
+            default:
+                break;
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:134]: wouldn't default mean that we receive some weird thing? should we throw an error instead?

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java [hunk 4]
```diff
@@ -68,8 +68,72 @@ public static DtmfTone fromString(String name) {
...
+            case "seven":
+                toneInChar = '7';
+                break;
+            case "eight":
+                toneInChar = '8';
+                break;
+            case "nine":
+                toneInChar = '9';
+                break;
+            case "a":
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:115]: are you sure about this being lowercase? to be 100% sure, just add a `toLowercase` call at the beginning.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java (hunk 2 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:73]: will remove the private embed function

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java (hunk 4 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:115]: good point, will translate to lower case

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java (hunk 3 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:134]: currently it return a empty char, will look at error throws

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CollectTonesResult.java (hunk 1 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CollectTonesResult.java]: it is a naming that the converted type is an element in char format, we could convert it to string directly, but the naming would be toString or convertToString, which doesn't reflect the motivation of the change purpose.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java (hunk 3 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:130]: throw exception cause a chains of problems that every place use this function would need exception handling. On the other hand, the DtmfTone is a result parsed from the returned schema, shouldn't the error be handled in the initial DtmfTone construction other than string converstion?

Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DtmfTone.java:130]: will make the default to return empty char.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CollectTonesResult.java (hunk 1 above)
Comment (cochi2) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CollectTonesResult.java]: > which doesn't reflect the motivation of the change purpose.
I thought that the change purpose is to create a string out of the CollectTonesResult object and `toChar` was just an intermediate operation. If you get rid of `toChar` and call directly `toString` you do the same work with one step less.

**Review by cochi2 — APPROVED** [review]

**Pull request merged**